File: Assets/PixelPlanets/StellarBreaker/Scripts/Economy/OfflineIncome.cs
using StellarBreaker.Core;

namespace StellarBreaker.Economy
{
    /// <summary>
    /// Estimates Stardust earned per second from idle combat:
    ///   kills/sec ≈ fleetDPS / enemyHP ;  income/sec = kills/sec × goldPerKill.
    /// Returns 0 if there is no passive DPS (idle needs ships).
    /// </summary>
    public static class OfflineIncome
    {
        public static BigNumber PerSecond(BigNumber fleetDps, BigNumber enemyHp, BigNumber goldPerKill)
        {
            if (fleetDps <= BigNumber.Zero || enemyHp <= BigNumber.Zero) return BigNumber.Zero;
            return (fleetDps / enemyHp) * goldPerKill;
        }
    }
}

File: TelegramApp/src/game/economy/OfflineIncome.ts
// Ported from Assets/PixelPlanets/StellarBreaker/Scripts/Economy/OfflineIncome.cs
import { BigNumber } from '../core/BigNumber'

/**
 * Estimates Stardust earned per second from idle combat:
 *   kills/sec ~ fleetDPS / enemyHP ;  income/sec = kills/sec * goldPerKill.
 * Returns 0 if there is no passive DPS (idle needs ships).
 */
export function offlineIncomePerSecond(fleetDps: BigNumber, enemyHp: BigNumber, goldPerKill: BigNumber): BigNumber {
  if (fleetDps.lte(BigNumber.Zero) || enemyHp.lte(BigNumber.Zero)) return BigNumber.Zero
  return fleetDps.div(enemyHp).mul(goldPerKill)
}
